test/entanglement_forging/test_entanglement_forging_knitter.py

```diff
@@ -101,7 +90,9 @@ def test_entanglement_forging_H2(self):
         forging_knitter = EntanglementForgingKnitter(ansatz)
 
         # Specify the decomposition method and get the forged operator
-        hamiltonian_terms, energy_shift = cholesky_decomposition(problem)
+        hamiltonian_terms, energy_shift = cholesky_decomposition(
+            problem, driver._calc.mo_coeff
```

Comment:
If you want to avoid accessing the private `_calc` attribute you could instead do the following:

1. replace `to_problem` with `to_qcschema` and store the resulting `QCSchema` object
2. use the methods provided in `qiskit_nature.second_q.formats`:
    - `qcschema_to_problem`: this is the same as done in `driver.to_problem`
    - `get_ao_to_mo_from_qcschema`: to build a `BasisTransformer` from the `QCSchema` object which has the `mo_coeff` publicly available
3. alternatively you can also extract the `mo_coeff` from the public attributes of the `QCSchema` object